libraries/opensk/src/api/connection.rs

```diff
@@ -46,7 +46,8 @@ pub struct SendOrRecvError;
 pub type SendOrRecvResult = Result<SendOrRecvStatus, SendOrRecvError>;
 
 pub trait HidConnection {
-    fn send_and_maybe_recv(&mut self, buf: &mut [u8; 64], timeout_ms: usize) -> SendOrRecvResult;
+    fn send(&mut self, buf: &[u8; 64], endpoint: UsbEndpoint) -> SendOrRecvResult;
+    fn recv(&mut self, buf: &mut [u8; 64], timeout_ms: usize) -> SendOrRecvResult;
```

Comment:
Shouldn't we have different responses now?
- Send returns Sent, Timeout?, or Error.
- Receive returns Received(UsbEndpoint), Timeout, or Error.

Reply:
This is currently just a convenience to not have 2 types, and the Send status is unreachable in `recv` and vice versa.
In our implementation, `send` can actually timeout, but that is an error, not a valid status like in the blocking `recv`. So `send` should return a:
`Result<(), SendOrRecvError>` (Do we want to specify the error more? I don't think any caller cares.)
and `recv` can just remove the Send status.

Reply:
It's not really a convenience to not have 2 types if every usage needs to have a panic branch when matching the wrong enum variant. And yes, having specific errors is probably useless. We should either use the CTAP errors or use a dummy unit type.

Reply:
Convenience in the sense of keeping this diff smaller for an initial review. But the new error types touched most of the files again, so turned out not too reviewer friendly. I hope its more readable now!